apps/ejabberd/src/jlib.erl

```diff
@@ -368,22 +363,14 @@ jid_to_binary({Node, Server, Resource}) ->
          end,
     S3.
 
--spec is_nodename([] | binary()) -> boolean().
+-spec is_nodename(<<>> | binary()) -> boolean().
 is_nodename([]) ->
```

Comment:
I changed this to empty binary. Empty list was a leftover after switching form lists to binary. This line was not covered by our tests and that's why we didn't catch it before.